File: code/typescript-for-csharp-java/errors/l01_order.ts
// errors/l01_order.ts
interface Line {
  product: string;
  price: number;
  quantity: number;
}

function withTax(line: Line): number {
  return line.price + line.price * 0.15;
}

// The prices come from a CSV file, where everything is text
const lines: Line[] = [
  { product: 'capo', price: '9.99', quantity: 2 },
  { product: 'strings', price: 12.5, quantity: 1 },
];
for (const line of lines) {
  console.log(line.product, withTax(line) * line.quantity);
}

File: code/python-for-csharp-java/errors/l01_order.py
# errors/l01_order.py
def with_tax(price: float) -> float:
    return price + price * 0.15


def line_total(product: str, price: float, quantity: int) -> float:
    if quantity > 10:
        return with_tax(price) * quantity * discount  # discount is defined nowhere
    return with_tax(price) * quantity


# The prices come from a CSV file, where everything is text
print("capo", line_total("capo", 9.99, 2))
print("strings", line_total("strings", "12.50", 1))
print("picks", line_total("picks", 0.5, 50))
